src/mscorlib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -1138,48 +1169,76 @@ private static TransitionTime TZif_CreateTransitionTimeFromPosixRule(string date
-                    {
-                        timeOfDay = new DateTime(1, 1, 2, 0, 0, 0);
-                    }
-                    else
-                    {
-                        timeOfDay = new DateTime(1, 1, 1, 0, 0, 0);
-                    }
-
-                    timeOfDay += timeOffset.Value;
-                }
-                else
+                return TransitionTime.CreateFloatingDateRule(ParseTimeOfDay(time), month, week, day);
+            }
+            else
+            {
+                // Julian day
+                int month, day;
+                if (!TZif_ParseJulianDay(date, out month, out day))
                 {
-                    // default to 2AM.
-                    timeOfDay = new DateTime(1, 1, 1, 2, 0, 0);
+                    throw new InvalidTimeZoneException(SR.InvalidTimeZone_InvalidJulianDay);
                 }
 
-                return TransitionTime.CreateFloatingDateRule(timeOfDay, month, week, day);
+                return TransitionTime.CreateFixedDateRule(ParseTimeOfDay(time), month, day);
             }
-            else
+        }
+
+        /// <summary>
+        /// Parses a string like Jn or n into month and day values.
+        /// </summary>
+        /// <returns>
+        /// true if the parsing succeeded; otherwise, false.
+        /// </returns>
+        private static bool TZif_ParseJulianDay(string date, out int month, out int day)
+        {
+            // Jn
+            // This specifies the Julian day, with n between 1 and 365.February 29 is never counted, even in leap years.
+            // n
+            // This specifies the Julian day, with n between 0 and 365.February 29 is counted in leap years.
+            Debug.Assert(!String.IsNullOrEmpty(date));
+            month = day = 0;
+
+            int index = 0;
+            bool isLeapYear = true;
+            if (date[index] == 'J')
+            {
+                isLeapYear = false;
+                index++;
+            }
+
+            if (index >= date.Length || ((uint)(date[index] - '0') > '9'-'0'))
             {
-                // Jn
-                // This specifies the Julian day, with n between 1 and 365.February 29 is never counted, even in leap years.
+                return false;
+            }
 
-                // n
-                // This specifies the Julian day, with n between 0 and 365.February 29 is counted in leap years.
+            int julianDay = 0;
 
-                // These two rules cannot be expressed with the current AdjustmentRules
-                // One of them *could* be supported if we relaxed the TransitionTime validation rules, and allowed
-                // "IsFixedDateRule = true, Month = 0, Day = n" to mean the nth day of the year, picking one of the rules above
+            do
+            {
+                julianDay = julianDay * 10 + (int) (date[index] - '0');
+                index++;
+            } while (index < date.Length && ((uint)(date[index] - '0') <= '9'-'0'));
```

Comment:
Is it not possible to use `int.TryParse` here?

Reply:
It is possible but my implementation should be more performant.